### PR TITLE
{bio}[foss/2022b] R-bundle-Bioconductor v3.16

### PR DESCRIPTION
**Pull request by branfosj**
(created using `eb --new-pr`)

* [x] #17200
* [x] #17565

### CODE REVIEW

File: easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.16-foss-2022b-R-4.2.2.eb [hunk 1]
```diff
@@ -0,0 +1,1240 @@
...
+homepage = 'https://bioconductor.org'
+description = """Bioconductor provides tools for the analysis and coprehension
+ of high-throughput genomic data."""
+
+toolchain = {'name': 'foss', 'version': '2022b'}
+
+# lpsymphony fails to build with pkgconf, so stick to pkg-config
+# error in: https://github.com/easybuilders/easybuild-easyconfigs/pull/15829#issuecomment-1178655286
+# builddependencies = [('pkg-config', '0.29.2')]
+builddependencies = [('pkgconf', '1.9.3')]
```
Comment (branfosj) [inline easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.16-foss-2022b-R-4.2.2.eb:16]: Testing if this has been fixed in `lpsymphony` yet...

File: easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.16-foss-2022b-R-4.2.2.eb [hunk 2]
```diff
@@ -0,0 +1,1240 @@
...
+
+# check whether correct version is installed in extension filter
+# (some versions in this bundle may be newer than the ones provided by R)
+local_ext_version_check = "pkgver = packageVersion('%(ext_name)s'); if (pkgver != '%(ext_version)s') "
+local_stop_msg = "stop('%(ext_name)s %(ext_version)s not installed, found ', pkgver, ' instead')"
+exts_filter = ("R -q --no-save", "%s { %s }" % (local_ext_version_check, local_stop_msg))
+
+# CRAN packages on which these Bioconductor packages depend are available in R module on which this depends
+# !! order of packages is important !!
+# packages updated on 7th Nov 2022
```
Comment (branfosj) [inline easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.16-foss-2022b-R-4.2.2.eb:49]:
```suggestion
# packages updated on 18th March 2023
```

File: easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.16-foss-2022b-R-4.2.2.eb [hunk 3]
```diff
@@ -0,0 +1,1240 @@
...
+    ('GEOquery', '2.66.0', {
+        'checksums': ['7602d0cb9c8d600187d599c2fb6d7328f219193c72c4af1987aaf51968556232'],
+    }),
+    ('rARPACK', '0.11-0', {
+        'checksums': ['c33401e2e31d272d485ce2ed22e7fe43ac641fd7c0a45a9b848d3ad60df1028a'],
+    }),
+    ('mixOmics', '6.22.0', {
+        'checksums': ['3dabec627b5e21da365979d6980565bd338e5a6ec8c3a016f3726ebb6184da70'],
+    }),
+    ('Rhdf5lib', '1.20.0', {
```
Comment (branfosj) [inline easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.16-foss-2022b-R-4.2.2.eb:327]: Failed to build ...

Comment (jfgrimm) [reply easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.16-foss-2022b-R-4.2.2.eb:327]: maybe it doesn't like HDF5 1.14? Might need to use the compat macros

Comment (branfosj) [reply easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.16-foss-2022b-R-4.2.2.eb:327]: > This R/Bioconductor package provides a mechanism to install the HDF5 system library via an R package.

I am not sure 1.14 is involved here. I fear it is installing an ancient HDF5 version that is the problem. See the repo at https://github.com/grimbough/Rhdf5lib

Comment (jfgrimm) [reply easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.16-foss-2022b-R-4.2.2.eb:327]: ugh
looks like it ships 1.10.7: https://github.com/grimbough/Rhdf5lib/blob/master/src/Makevars.in

File: easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.16-foss-2022b-R-4.2.2.eb [hunk 4]
```diff
@@ -0,0 +1,1240 @@
...
+        'checksums': ['9c78255ca476c945c05a564d1e4da363de714d890e0e27f3b252fd73c50eed71'],
+    }),
+    ('Rfast', '2.0.7', {
+        'checksums': ['8f86159a4760a7124e1c91ae0b022c7e496f81590ea4e4af702bea44e8dedf8f'],
+    }),
+    ('ALDEx2', '1.30.0', {
+        'checksums': ['6eba6fdd689fdd5d18fde3ffe4aa64ad4af5bfdc8977b8038d297fb46ed10515'],
+    }),
+    ('dada2', '1.26.0', {
+        'patches': ['dada2-1.16.0-remove-sse-on-aarch64.patch'],
```
Comment (branfosj) [inline easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.16-foss-2022b-R-4.2.2.eb:445]: Patch does not apply cleanly - possibly no longer needed

File: easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.16-foss-2022b-R-4.2.2.eb (hunk 3 above)
Comment (branfosj) [reply easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.16-foss-2022b-R-4.2.2.eb:327]: Argh. This was a mistake by me. I'd broken my cURL install and this caused the problems here.

File: easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.16-foss-2022b-R-4.2.2.eb (hunk 4 above)
Comment (branfosj) [reply easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.16-foss-2022b-R-4.2.2.eb]: Patch is no longer needed - https://benjjneb.github.io/dada2/ReleaseNotes_1_26.html

**Comment by branfosj** [comment]
@boegelbot please test @ generoso

File: easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.16-foss-2022b-R-4.2.2.eb (hunk 1 above)
Comment (branfosj) [reply easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.16-foss-2022b-R-4.2.2.eb]: Building for me ...

**Comment by branfosj** [comment]
@boegelbot please test @ jsc-zen2

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0105u03a.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/5888a5f7d17a9532d9e6522e60fc8a58 for a full test report.

**Comment by smoors** [comment]
> WIP while I test it

is this still wip?

**Comment by branfosj** [comment]
> > WIP while I test it
> 
> is this still wip?

No, it is ready to test.

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
easybuild-c7.novalocal - Linux CentOS Linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/verdurin/5888228bea2c8f15e07419a20f7bffd3 for a full test report.

**Comment by verdurin** [comment]
@boegelbot please test @ jsc-zen2

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3150.skitty.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/boegel/dbf5780317c926ae427369cc2d83c908 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @branfosj!

**Pull request merged**